### DISCUSSION TITLE
Disable gif autoplay

### DISCUSSION THREAD

**Discussion by ghost** [forem/forem#15486] [Completed]
**Is your feature request related to a problem? Please describe.**

For accessibility reasons and to improve the reader experience I would like to have the option to disable all animations embedded into dev.to pages. For example, I find animated gifs extremely distracting from the content of the page.

**Describe the solution you'd like**
I would like an option ~in a user setting~ to disable/enable animations. ~Then when I visit a page that has animated gif, they would not play and I would just see a still image.~ This would allow users to click on a gif to stop it from auto playing. In the future, having a setting for this would be great but that will require a full RFC.

**Describe alternatives you've considered**
With Chrome there are Chrome extensions for this, but would prefer to control it via the dev.to website.

**Comment by Zhao-Andy** [comment]
Hey @ryanwhocodes, thanks for the issue. I think this makes sense from an accessibility standpoint.

I think the implementation would be something like a user column, and then we can check that column before rendering images. Not sure how complicated it would be though. Any thoughts @benhalpern?

**Comment by ghost** [comment]
From a small bit of research so far, some suggestions include having a static gif saved as well as an animated one, and which one is delivered to the user depends on the user's setting.

**Comment by jessleenyc** [comment]
Bumping this up for you @benhalpern. I think we should approve this.

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by abraham** [comment]
There is an newish media query that would be perfect for this. It's already shipped in Firefox, Safari, and Chrome 74 (currently dev).

https://developers.google.com/web/updates/2019/03/prefers-reduced-motion

**Comment by benhalpern** [comment]
Oh cool. Our new config options should make individual preferences more attainable.

How we architect our configuration will affect how easy it is to add long tail config like this.

**Comment by smmaci** [comment]
A potentially simpler solution in the meantime would be to allow users to close/delete/remove the animated element with some sort of control, like an "x" icon at the top of the image. 

Personally, I don't mind seeing an animation play once, but I can't focus when they are repeated.

**Comment by jessleenyc** [comment]
Maybe this is the solution:https://github.com/thepracticaldev/dev.to/issues/3638
On Twitter, you can click on the "gif" and it'll stop playing.

**Comment by cmgorton** [comment]
@forem/frontend want to put this issue on your radar. It has been around for awhile. Wondering if we should add this to "issues to review" for the planning funnel or if you think we should close this. I think it is an important consideration for users.

**Comment by aitchiss** [comment]
@cmgorton I'd be keen to add it to "issues to review" - I agree it's an important consideration and there's a couple of solid potential approaches already mentioned above.

Being able to pause/stop/hide animating content is required in the [WCAG success criteria](https://www.w3.org/TR/WCAG21/#pause-stop-hide) minimum standard and definitely something we should be aiming for

**Comment by vaidehijoshi** [comment]
@cmgorton + @aitchiss This is definitely a good issue for us to add to the funnel (especially given the fact that no one from the community has picked it up and it has been open for some time). I've gone ahead and added it! 🚀

**Comment by cmgorton** [comment]
Great! Thanks Suzanne and Vaidehi🥳

**Comment by mstruve** [comment]
@aitchiss the user here asks for a setting and that would throw this in the RFC category. However, we would love to make this more accessible immediately if possible. Is there a way to simply pause GIFs by clicking on them like Jess alluded to in her linked issue? If so we could update that to be the scope of the issue and prioritize this away

**Comment by aitchiss** [comment]
Yes we could do that, and I think having both options eventually (global preferences setting, and the individual play/pause) would be great.

When we show a post we would need to be able to fetch a static `jpg`/`png` of any gifs used in the content to display as an alternative to the animated gif. there's [a nice example of how we could switch out the content on CSS Tricks](https://css-tricks.com/pause-gif-details-summary/). 

It would be great if we also paused gifs by default if a user has 'prefers reduced motion' enabled in their browser settings (it's just a CSS media query)

**Comment by mstruve** [comment]
Updated the description of the Issue to make this a more straight forward fix and remove the need for an RFC

**Comment by InHuOfficial** [comment]
I wrote a quick article (on dev.to of course 😋) on a "hack" that could solve this temporarily (or possibly be developed into a solution for external links to GIFs as I can imagine that would be a nightmare to fix!).

It would need a bit of work before going into production and testing but I thought I would share it as this issue is a tough one to fix with user content, especially as they can link to external GIFs and even worse external webp images.

[Pausing GIF animations on dev.to for those who `prefer-reduced-motion`](https://dev.to/inhuofficial/stopping-gif-animations-on-dev-to-for-those-who-prefer-reduced-motion-59pg)

I don't use github much so if you want me my dev.to profile is [https://dev.to/inhuofficial](https://dev.to/inhuofficial) so you can message me there or drop a reply on the linked article.

**Comment by JennieOcken** [comment]
This was completed here: https://github.com/forem/forem/pull/16314
